### PR TITLE
Feature/zoom box

### PR DESCRIPTION
**Pull request by dvantetering**
- Create snapshot of the graph, which will be the background of the ZoomBox.
- Re-Introduce ZoomBox
- ZoomBox is moveable, but still has to be connected to the ZoomingLevels of the Graph
- ZoomBox is moveable with A (to the left) and D (to the right)

### CODE REVIEW

File: src/main/java/application/controllers/GraphController.java [hunk 1]
```diff
@@ -20,9 +26,13 @@
 @SuppressWarnings("PMD.UnusedPrivateField")
 public class GraphController extends Controller<ScrollPane> {
     private Graph graph;
-    private GraphMouseHandling graphMouseHandling;
+    //private GraphMouseHandling graphMouseHandling;
```
Comment (arbreurkes) [inline src/main/java/application/controllers/GraphController.java]: commented code should be omitted

File: src/main/java/application/controllers/GraphController.java [hunk 2]
```diff
@@ -33,7 +43,9 @@
     public GraphController(Graph g, Object ref) {
         super(new ScrollPane());
         this.graph = g;
-        this.graphMouseHandling = new GraphMouseHandling();
+        this.zoomController = new ZoomController();
+        this.maxWidth = 0;
+        //this.graphMouseHandling = new GraphMouseHandling();
```
Comment (arbreurkes) [inline src/main/java/application/controllers/GraphController.java]: Commented code should be omitted

File: src/main/java/application/controllers/GraphController.java [hunk 3]
```diff
@@ -73,26 +86,28 @@ public void initialize(URL location, ResourceBundle resources) {
...
-        System.out.println("init");
         AnchorPane root = new AnchorPane();
         graph.addGraphComponents(ref);
 
         // add components to graph pane
         root.getChildren().addAll(graph.getModel().getAddedEdges());
         root.getChildren().addAll(graph.getModel().getAddedCells());
 
-        graph.getModel().getAddedCells().forEach(graphMouseHandling::setMouseHandling);
+       // graph.getModel().getAddedCells().forEach(graphMouseHandling::setMouseHandling);
```
Comment (arbreurkes) [inline src/main/java/application/controllers/GraphController.java]: Commented code should be omitted

File: src/main/java/application/controllers/GraphController.java [hunk 4]
```diff
@@ -73,26 +86,28 @@ public void initialize(URL location, ResourceBundle resources) {
...
         root.getChildren().removeAll(graph.getModel().getRemovedEdges());
 
         graph.endUpdate();
-        CellLayout layout = new BaseLayout(graph.getModel(), 20,
+        BaseLayout layout = new BaseLayout(graph.getModel(), 20,
                 (int) (screenSize.getHeight() - 25) / 2);
         layout.execute();
-
+        maxWidth = layout.getMaxWidth();
+        System.out.println(maxWidth);
```
Comment (arbreurkes) [inline src/main/java/application/controllers/GraphController.java]: System.out.Prints should be omitted.

File: src/main/java/application/controllers/GraphController.java [hunk 5]
```diff
@@ -103,4 +118,16 @@ public void init(Object ref) throws IOException {
     public List<String> getGenomes() {
         return graph.getGenomes();
     }
+
+    public void takeSnapshot() throws IOException {
+        WritableImage image = new WritableImage((int)maxWidth + 50, (int) screenSize.getHeight());
+        WritableImage snapshot = this.getRoot().getContent().snapshot(new SnapshotParameters(), image);
+
+        System.out.println("take snapshot");
```
Comment (arbreurkes) [inline src/main/java/application/controllers/GraphController.java]: System.out.Prints should be omitted.

File: src/main/java/application/controllers/GraphController.java [hunk 6]
```diff
@@ -103,4 +117,14 @@ public void init(Object ref, int depth) throws IOException {
     public List<String> getGenomes() {
         return graph.getGenomes();
     }
+
+    public void takeSnapshot() throws IOException {
```
Comment (ghost) [inline src/main/java/application/controllers/GraphController.java:128]: Missing JDoc

File: src/main/java/application/controllers/ZoomController.java [hunk 7]
```diff
@@ -49,6 +47,7 @@ public KeyHandler getKeyHandler() {
         return keyHandler;
     }
 
+    public ZoomBox getZoomBox() { return zoomBox; }
```
Comment (ghost) [inline src/main/java/application/controllers/ZoomController.java:50]: Missing JDoc

File: src/main/java/application/fxobjects/graph/cell/BaseLayout.java [hunk 8]
```diff
@@ -178,4 +189,6 @@ public int getCellCount() {
     public void setCellCount(int cellCount) {
         this.cellCount = cellCount;
     }
+
+    public double getMaxWidth() { return maxWidth; }
```
Comment (ghost) [inline src/main/java/application/fxobjects/graph/cell/BaseLayout.java:193]: Missing JDoc

File: src/main/java/application/controllers/MainController.java [hunk 9]
```diff
@@ -104,6 +104,13 @@ public void fillGraph(Object ref) {
...
         this.getRoot().setCenter(screen);
         list.setItems(FXCollections.observableArrayList(graphController.getGenomes()));
+        this.getRoot().setBottom(graphController.getZoomController().getZoomBox().getZoomBox());
+
+//        try {
+//            graphController.takeSnapshot();
+//        } catch (IOException e) {
+//            e.printStackTrace();
+//        }
     }
```
Comment (arbreurkes) [inline src/main/java/application/controllers/MainController.java:153]: Commented code should be omitted.

File: src/main/java/application/fxobjects/graph/cell/BaseLayout.java [hunk 10]
```diff
@@ -42,6 +43,8 @@ public BaseLayout(Model model, int offset, int middle) {
         this.centerY = middle;
         //System.out.println(centerY + "   center Y");
```
Comment (arbreurkes) [inline src/main/java/application/fxobjects/graph/cell/BaseLayout.java]: Commented System out print.

File: src/main/java/application/controllers/GraphController.java [hunk 11]
```diff
@@ -97,16 +98,16 @@ public void init(Object ref, int depth) throws IOException {
...
         graph.endUpdate();
         BaseLayout layout = new BaseLayout(graph.getModel(), 20,
                 (int) (screenSize.getHeight() - 25) / 2);
         layout.execute();
-        maxWidth = layout.getMaxWidth();
+        maxWidth = (int) layout.getMaxWidth();
         this.getRoot().setContent(root);
 
-        takeSnapshot();
+        //takeSnapshot();
```
Comment (arbreurkes) [inline src/main/java/application/controllers/GraphController.java:110]: commented code should be omitted.

File: src/main/java/application/controllers/GraphController.java [hunk 12]
```diff
@@ -119,10 +120,11 @@ public void init(Object ref, int depth) throws IOException {
     }
 
     public void takeSnapshot() throws IOException {
-        WritableImage image = new WritableImage((int)maxWidth + 50, (int) screenSize.getHeight());
-        WritableImage snapshot = this.getRoot().getContent().snapshot(new SnapshotParameters(), image);
+        SnapshotParameters snapshotParameters = new SnapshotParameters();
+        //WritableImage image = new WritableImage((int)maxWidth + 50, (int) screenSize.getHeight());
```
Comment (arbreurkes) [inline src/main/java/application/controllers/GraphController.java:130]: more commented code.

File: src/main/java/application/controllers/MainController.java [hunk 13]
```diff
@@ -32,17 +32,13 @@
     private ScrollPane screen;
     @FXML
     private MenuBar menuBar;
-    @FXML
```
Comment (arbreurkes) [inline src/main/java/application/controllers/MainController.java:34]: Well noticed!

File: src/main/java/application/controllers/MainController.java [hunk 14]
```diff
@@ -115,8 +111,9 @@ private void createList() {
      */
     private void createNodeInfo() {
         infoList = new TextFlow();
-        infoList.prefHeightProperty().bind(infoScroller.heightProperty());
-        infoList.prefWidthProperty().bind(infoScroller.widthProperty());
+//        infoList.prefHeightProperty().bind(infoScroller.heightProperty());
+//        infoList.prefWidthProperty().bind(infoScroller.widthProperty());
```
Comment (arbreurkes) [inline src/main/java/application/controllers/MainController.java:115]: commented code

File: src/main/java/application/controllers/MainController.java [hunk 15]
```diff
@@ -144,8 +141,10 @@ public void fillGraph(Object ref) {
         screen = graphController.getRoot();
 
         this.getRoot().setCenter(screen);
+        System.out.println(screen);
```
Comment (arbreurkes) [inline src/main/java/application/controllers/MainController.java:144]: system out print. (should be omitted)

**Pull request merged**